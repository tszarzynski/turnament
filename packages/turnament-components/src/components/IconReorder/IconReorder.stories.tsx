import React from 'react';
import IconReorder from './IconReorder';

export default {
  component: IconReorder,
  title: 'IconReorder',
};

export const Default = () => {
  return <IconReorder />;
};