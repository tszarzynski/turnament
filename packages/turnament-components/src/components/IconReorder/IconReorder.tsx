const IconReorder = () => {
	return (
		<svg
			className=" w-10 h-10"
			xmlns="http://www.w3.org/2000/svg"
			width="24"
			height="24"
			viewBox="0 0 24 24"
			fill="none"
			stroke="currentColor"
			strokeWidth="1"
			strokeLinecap="round"
			strokeLinejoin="round"
			aria-label="Reorder icon"
			role="img"
		>
			<path stroke="none" d="M0 0h24v24H0z" />{" "}
			<path d="M3 9l4-4l4 4m-4 -4v14" /> <path d="M21 15l-4 4l-4-4m4 4v-14" />
		</svg>
	);
};

export default IconReorder;
