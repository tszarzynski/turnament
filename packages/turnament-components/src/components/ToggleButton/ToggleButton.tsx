import type React from "react";

type Props = {
	checked: boolean;
	onChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
	iconSlot: React.ReactNode;
	shape?: "square" | "circle";
	name?: string;
	disabled?: boolean;
	className?: string;
	title?: string;
};

const ToggleButton = ({
	iconSlot,
	checked,
	onChange,
	disabled,
	className,
	name,
	title,
	shape = "square",
}: Props) => {
	const checkedStyles = checked ? "text-primary" : "text-secondary";

	const disabledStyles =
		"border-gray-300 text-gray-300 cursor-not-allowed hover:bg-transparent";

	const shapeStyles = shape === "circle" ? "rounded-full" : "";

	return (
		<label
			className={`inline-flex items-center cursor-pointer focus-within:outline-none focus-within:ring-2 focus-within:ring-offset-2 ${checkedStyles} ${className}`}
		>
			<input
				type="checkbox"
				className="sr-only peer"
				checked={checked}
				onChange={disabled ? undefined : onChange}
				disabled={disabled}
				name={name}
			/>
			<div
				className={`border hover:opacity-50 transition-opacity ${shapeStyles}   ${disabled ? disabledStyles : checkedStyles}`}
				title={title}
			>
				{iconSlot}
			</div>
		</label>
	);
};

export default ToggleButton;
