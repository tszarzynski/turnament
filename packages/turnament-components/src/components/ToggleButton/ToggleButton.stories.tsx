import { useToggle } from "react-use";
import IconReorder from "../IconReorder";
import ToggleButton from "./ToggleButton";

export default {
	component: ToggleButton,
	title: "ToggleButton",
};

export const Default = () => {
	const [checked, toggle] = useToggle(false);
	return (
		<ToggleButton
			iconSlot={<IconReorder />}
			checked={checked}
			onChange={toggle}
		/>
	);
};

export const Checked = () => {
	const [checked, toggle] = useToggle(true);
	return (
		<ToggleButton
			iconSlot={<IconReorder />}
			checked={checked}
			onChange={toggle}
		/>
	);
};

export const Disabled = () => {
	const [checked, toggle] = useToggle(true);
	return (
		<ToggleButton
			iconSlot={<IconReorder />}
			checked={checked}
			onChange={toggle}
			disabled={true}
		/>
	);
};

export const Circle = () => {
	const [checked, toggle] = useToggle(false);
	return (
		<ToggleButton
			iconSlot={<IconReorder />}
			checked={checked}
			onChange={toggle}
			shape="circle"
		/>
	);
};
