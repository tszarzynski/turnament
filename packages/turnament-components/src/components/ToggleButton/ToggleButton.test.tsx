import { render } from "vitest-browser-react";
import IconReorder from "../IconReorder";
import ToggleButton from "./ToggleButton";

describe("ToggleButton", () => {
	it("renders without error", () => {
		const { asFragment } = render(
			<ToggleButton
				checked={true}
				onChange={() => undefined}
				iconSlot={<IconReorder />}
			/>,
		);
		expect(asFragment()).toMatchSnapshot();
	});
});
