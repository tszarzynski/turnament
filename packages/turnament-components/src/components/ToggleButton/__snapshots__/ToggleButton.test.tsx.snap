// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ToggleButton > renders without error 1`] = `
<DocumentFragment>
  <label
    class="inline-flex items-center cursor-pointer focus-within:outline-none focus-within:ring-2 focus-within:ring-offset-2 text-primary undefined"
  >
    <input
      checked=""
      class="sr-only peer"
      type="checkbox"
    />
    <div
      class="border hover:opacity-50 transition-opacity    text-primary"
    >
      <svg
        aria-label="Reorder icon"
        class=" w-10 h-10"
        fill="none"
        height="24"
        role="img"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="1"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0z"
          stroke="none"
        />
         
        <path
          d="M3 9l4-4l4 4m-4 -4v14"
        />
         
        <path
          d="M21 15l-4 4l-4-4m4 4v-14"
        />
      </svg>
    </div>
  </label>
</DocumentFragment>
`;
